gammapy/cube/core.py

```diff
@@ -312,21 +312,15 @@ def to_sherpa_data3d(self):
     @property
     def solid_angle_image(self):
         """Solid angle image in steradian (`~astropy.units.Quantity`)"""
-        cube_hdu = fits.ImageHDU(self.data, self.wcs.to_header())
+        skymap = self.sky_image()
+        return skymap.solid_angle()
 
-        from .utils import cube_to_image
-        image_hdu = cube_to_image(cube_hdu)
-        image_hdu.header['WCSAXES'] = 2
-
-        sky_map = SkyMap.from_image_hdu(image_hdu)
-        return sky_map.solid_angle()
-
-    def sky_image(self, idx_energy, copy=True):
+    def sky_image(self, idx_energy=None, copy=True):
         """Slice 3-dim cube into a 2-dim `~gammapy.image.SkyMap` along slicepos.
 
         Parameters
         ----------
-        inx_energy : int
+        idx_energy : int or None (default None)
```

Comment:
I'm not sure if adding this `None` option to sum over the energy axis is a good idea.
We should have (and partly already do have) separate methods to create such "projections".
So my suggestion here would be to remove that option again, and in `solid_angle_image` just call `sky_image(idx_energy=0)`.